docs/template_plugin/src/template_infer_request.cpp

```diff
@@ -64,46 +66,27 @@ static void AllocateImpl(const BlobDataMap& userDataMap,
                          GetNetworkPrecisionF&& GetNetworkPrecision,
                          bool isInputBlob = true) {
     for (auto&& userData : userDataMap) {
-        auto& dims = userData.second->getTensorDesc().getDims();
-        const auto devicePrecision = Precision::FP32;
+        const auto& dims = userData.second->getTensorDesc().getDims();
         const auto deviceLayout = TensorDesc::getLayoutByDims(dims);
-        auto userPrecision = userData.second->getTensorDesc().getPrecision();
-        auto userLayout = userData.second->getTensorDesc().getLayout();
+        const auto userPrecision = userData.second->getTensorDesc().getPrecision();
+        const auto userLayout = userData.second->getTensorDesc().getLayout();
 
-        Blob::Ptr userBlob;
-        switch (userPrecision) {
-            case Precision::U8: {
-                userBlob = InferenceEngine::make_shared_blob<std::uint8_t>({userPrecision, dims, userLayout});
-            } break;
-            case Precision::FP32 : {
-                userBlob = InferenceEngine::make_shared_blob<float>({userPrecision, dims, userLayout});
-            } break;
-            default: IE_THROW(NotImplemented) << "Template Plugin: Unsupported Input/Output Precision";
-        }
+        const auto networkPrecision = InferenceEngine::details::convertPrecision(GetNetworkPrecision(userData.first));
+        Blob::Ptr userBlob = make_blob_with_precision({userPrecision, dims, userLayout});
         userBlob->allocate();
         userBlobMap[userData.first] = userBlob;
 
-        auto networkPrecision = GetNetworkPrecision(userData.first);
         Blob::Ptr deviceBlob;
-        switch (networkPrecision) {
-            case ngraph::element::Type_t::f32 : {
-                if (userPrecision == devicePrecision && userLayout == deviceLayout) {
-                    deviceBlob = userBlob;
-                } else {
-                    deviceBlob = InferenceEngine::make_shared_blob<float>({devicePrecision, dims, deviceLayout});
-                }
-            } break;
-            default: IE_THROW(NotImplemented) << "Template Plugin: Unsupported network Input/Output Presision";
-        }
-        if (userBlob != deviceBlob) {
-            if (isInputBlob) {
-                // preprocessing converts user input blob to desired device input blob automatically
-                deviceBlob->allocate();
-            } else {
-                // NOTE: this is not supported for output user blobs yet
-                IE_THROW(NotImplemented) << "Template Plugin: does not support setPrecision, setLayout for outputs";
+        if (userPrecision == networkPrecision && userLayout == deviceLayout) {
```

Comment:
Can we compare tensor description here directly?

Reply:
we don't have tensor desc which represents network precision and device layout, they are extract using different paths